pandas/tests/indexes/test_datetimelike.py

```diff
@@ -732,6 +741,13 @@ def test_fillna_datetime64(self):
                            dtype=object)
             self.assert_index_equal(idx.fillna('x'), exp)
 
+    def test_difference_freq(self):
+        # GH14323. difference of DatetimeIndex does not preserve frequency,
+        # so a differencing operation should not retain the freq field of the
+        # original index.
+        i = self.create_index()
+        self.assertIsNone(i.difference(i[1:3]).freq)
```

Comment:
> assert_index_equal looks like it only compares freq for PeriodIndex

Indeed, that seems the case. So it is ok to still explicitly check the freq attribute to be None of the result

> I can't use the self.indices field because the point is I need an index result that doesn't have a regular freq.

But the starting index should have a freq, no? So you can check the freq is reset to None after doing a difference?